pkg/kv/kvserver/batcheval/cmd_gc.go

```diff
@@ -44,6 +44,14 @@ func declareKeysGC(
 			latchSpans.AddMVCC(spanset.SpanReadWrite, roachpb.Span{Key: key.Key}, header.Timestamp)
 		}
 	}
+	for _, rKey := range gcr.RangeKeys {
+		// Extend the range key latches by feather to ensure MVCC stats
+		// computations correctly account for adjacent ranges if then need to be
+		// split.
+		l, r := rangeTombstonePeekBounds(rKey.StartKey, rKey.EndKey, rs.GetStartKey().AsRawKey(), nil)
+		latchSpans.AddMVCC(spanset.SpanReadWrite, roachpb.Span{Key: l, EndKey: r},
```

Comment:
These latches are a bit iffy.

We do need to take out an exclusive write latch across all timestamps, because a future write could fragment a past range tombstone, causing incorrect stats updates -- so we handle that here. When we start clearing large amounts of garbage using `ClearRawRange` we'll also need that exclusive write latch.

However, we use `header.Timestamp` here, which excludes future readers. I don't believe there's any need for that, because after #78980 reads operate on a stable Pebble snapshot taken after checking the GC threshold.

I think ideally we would want this to be a write-only latch, and serialize with other GC requests via some shared key like `RangeLastGCKey` (or alternatively some variant of read-latches that affect past readers only). Do we support write-only latches currently @nvanbenschoten? We could also add `SpanWriteOnly`, or maybe hack around it with passing `MaxTimestamp`.

Also, taking out these wide write latches is going to be pretty bad if they're held for any amount of time, so we need to make sure we make these requests very cheap, and be mindful of their effect on request queues. We can revisit these optimization concerns later, but it'd be good to have a `TODO` for it.